src/main/cli/cli.c

```diff
@@ -1362,6 +1362,66 @@ static void cliSerialPassthrough(char *cmdline)
 }
 #endif
 
+#if defined(USE_ESC_SENSOR)
+static void cliEscSensorPassthrough(char *cmdline)
+{
+    uint32_t useEscSensorBaud = 115200;
+    if (!isEmpty(cmdline)) {
+        useEscSensorBaud = atoi(cmdline);
+    }
+    
+    const serialPortConfig_t *portConfig = findSerialPortConfig(FUNCTION_ESC_SENSOR);
+    if (!portConfig) {
+        cliPrintLine("no esc sensor port configured.");
+        return;
+    }
+    
+    serialPortUsage_t *serialPortUsage = findSerialPortUsageByIdentifier(portConfig->identifier);
+    serialPortUsage->function = FUNCTION_NONE;
+    serialPortUsage->serialPort = NULL;
+    
+    serialPort_t *passThroughPort;
+    passThroughPort = openSerialPort(portConfig->identifier,
+        FUNCTION_NONE,
+        NULL,
+        NULL,
+        useEscSensorBaud,
+        MODE_RXTX,
+        (SERIAL_STOPBITS_1 | SERIAL_PARITY_NO | SERIAL_BIDIR | SERIAL_BIDIR_PP)
+    );
+    
+    if (!passThroughPort) {
+        cliPrintLine("esc sensor port could not be opened.");
+        return;
+    }
+    
+    pwmDisableMotors();
+    pwmOutputPort_t *pwmMotors = pwmGetMotors();
+    for (volatile uint8_t i = 0; i < MAX_SUPPORTED_MOTORS; i++) {
+        if (pwmMotors[i].enabled && pwmMotors[i].io != IO_NONE) {
+            const ioTag_t tag = motorConfig()->dev.ioTags[i];
+            if (tag != IO_TAG_NONE) {
+                const timerHardware_t *timerHardware = timerGetByTag(tag);
+                if (timerHardware) {
+                    IOInit(pwmMotors[i].io, OWNER_MOTOR, 0);
+                    IOConfigGPIO(pwmMotors[i].io, IOCFG_OUT_PP);
+                    if (timerHardware->output & TIMER_OUTPUT_INVERTED) {
+                        IOLo(pwmMotors[i].io);
+                    }else{
+                        IOHi(pwmMotors[i].io);
+                    }
+                }
+            }
+        }
+    }
+    
+    cliPrintLine("esc sensor passthrough enabled, power cycle to exit.");
```

Comment:
Forcing the user to physically disconnect power is a poor solution for this. This should allow dropping of DTR to be used for a reset as well.

Reply:
i just took it from other cli commands using serialPassthrough(). dont know if a DTR will still work